lib/vector-config-common/src/schema_gen/schema.rs

```diff
@@ -0,0 +1,563 @@
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub format: Option<String>,
+
+    /// The `enum` keyword.
+    ///
+    /// See [JSON Schema Validation 6.1.2. "enum"](https://tools.ietf.org/html/draft-handrews-json-schema-validation-02#section-6.1.2)
+    #[serde(rename = "enum", skip_serializing_if = "Option::is_none")]
+    pub enum_values: Option<Vec<Value>>,
+
+    /// The `const` keyword.
+    ///
+    /// See [JSON Schema Validation 6.1.3. "const"](https://tools.ietf.org/html/draft-handrews-json-schema-validation-02#section-6.1.3)
+    #[serde(
+        rename = "const",
+        skip_serializing_if = "Option::is_none",
+        deserialize_with = "allow_null"
+    )]
+    pub const_value: Option<Value>,
+
+    /// Properties of the [`SchemaObject`] which define validation assertions in terms of other schemas.
+    #[serde(flatten, deserialize_with = "skip_if_default")]
+    pub subschemas: Option<Box<SubschemaValidation>>,
+
+    /// Properties of the [`SchemaObject`] which define validation assertions for numbers.
+    #[serde(flatten, deserialize_with = "skip_if_default")]
+    pub number: Option<Box<NumberValidation>>,
+
+    /// Properties of the [`SchemaObject`] which define validation assertions for strings.
+    #[serde(flatten, deserialize_with = "skip_if_default")]
+    pub string: Option<Box<StringValidation>>,
+
+    /// Properties of the [`SchemaObject`] which define validation assertions for arrays.
+    #[serde(flatten, deserialize_with = "skip_if_default")]
+    pub array: Option<Box<ArrayValidation>>,
+
+    /// Properties of the [`SchemaObject`] which define validation assertions for objects.
+    #[serde(flatten, deserialize_with = "skip_if_default")]
+    pub object: Option<Box<ObjectValidation>>,
+
+    /// The `$ref` keyword.
+    ///
+    /// See [JSON Schema 8.2.4.1. Direct References with "$ref"](https://tools.ietf.org/html/draft-handrews-json-schema-02#section-8.2.4.1).
+    #[serde(rename = "$ref", skip_serializing_if = "Option::is_none")]
+    pub reference: Option<String>,
+
+    /// Arbitrary extra properties which are not part of the JSON Schema specification, or which `schemars` does not support.
+    #[serde(flatten)]
+    pub extensions: Map<String, Value>,
+}
+
+// Deserializing "null" to `Option<Value>` directly results in `None`,
+// this function instead makes it deserialize to `Some(Value::Null)`.
+fn allow_null<'de, D>(de: D) -> Result<Option<Value>, D::Error>
+where
+    D: serde::Deserializer<'de>,
+{
+    Value::deserialize(de).map(Option::Some)
+}
+
+fn skip_if_default<'de, D, T>(deserializer: D) -> Result<Option<Box<T>>, D::Error>
+where
+    D: serde::Deserializer<'de>,
+    T: Deserialize<'de> + Default + PartialEq,
+{
+    let value = T::deserialize(deserializer)?;
+    if value == T::default() {
+        Ok(None)
+    } else {
+        Ok(Some(Box::new(value)))
+    }
```

Comment:
I know you're just copying from source, but consider:
```suggestion
    Ok((value != T::default()).then(|| Box::new(value)))
```

Reply:
Definitely doesn't hurt to make improvements, even as a copypasta job.

Fixed in 27beb14e2.